examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls

```diff
@@ -30,12 +30,20 @@ global class B2CDeliverySample implements sfdc_checkout.CartShippingCharges {
             }
 
             // On re-entry of the checkout flow delete all previous CartDeliveryGroupMehods for the given cartDeliveryGroupId
-            delete [SELECT Id FROM CartDeliveryGroupMethod WHERE CartDeliveryGroupId = :cartDeliveryGroupId];
+            delete [SELECT Id FROM CartDeliveryGroupMethod WHERE WebCartId = :cartId];
 
             // Create a CartDeliveryGroupMethod record for every shipping option returned from the external service
+            Integer cdgmToBeCreated = 0;
             for (ShippingOptionsAndRatesFromExternalService shippingOption: shippingOptionsAndRatesFromExternalService) {
-               populateCartDeliveryGroupMethodWithShippingOptions(shippingOption, cartDeliveryGroupId, cartId);
+                for(CartDeliveryGroup curCartDeliveryGroup : cartDeliveryGroups){
+                    populateCartDeliveryGroupMethodWithShippingOptions(shippingOption, curCartDeliveryGroup.Id, cartId);
```

Comment:
Here before populating the cartDeliveryGroupMethods for a deliveryGroup, we need to check if there are any cartItems in the deliveryGroup, else we will have ShippingMethods for an emptyDeliveryGroup which is confusing